File: microservices/business/user-microservice/user-microservice-out-mongo-port/src/main/java/com/joserbatista/service/user/outbound/mongo/document/UserDocument.java
package com.joserbatista.service.user.outbound.mongo.document;

import lombok.Builder;
import lombok.Data;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.UUID;

@Document("users")
@Data
@Builder
public class UserDocument {

    @Id
    private String documentId;

    @Builder.Default
    @Indexed(unique = true)
    private String id = UUID.randomUUID().toString();

    @Indexed(unique = true)
    private String username;
    @ToString.Exclude
    private String password;

    @Indexed(unique = true)
    private String email;

    private String fullName;

    private Boolean enabled;
}

File: microservices/_cloud/gateway-microservice/src/main/kotlin/com/joserbatista/cloud/gatewaymicroservice/auth/UserDocument.kt
package com.joserbatista.cloud.gatewaymicroservice.auth

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document("users")
data class UserDocument(

    @field:Id
    val documentId: String? = null,

    @field:Indexed(unique = true)
    val id: String = UUID.randomUUID().toString(),

    @field:Indexed(unique = true)
    val username: String? = null,

    val password: String? = null,

    val enabled: Boolean? = null,
)